### DISCUSSION TITLE
time triggers are one hour off

### DISCUSSION THREAD

**Discussion by Danahi** [custom-components/pyscript#312] [Q&A]
My time triggers fire one hour later than they should.
I checked in home assistant what my timezone is set to and it is on my local timezone (Amsterdam).
Please help me 😨

**Comment by craigbarratt** [comment]
Pyscript just uses local time, without timezone.  What happens when you run this command:
```
python -c 'import datetime; print(datetime.datetime.now())'
```
You should use the same python version that HA is running.  Does it print the correct local time?

**Reply by Danahi** [reply]
Where can I run python commands? Or do you mean in a pyscript?

I didn't select any python version afaik, I just installed pyscript as HACS integration.

**Comment by Danahi** [comment]
Sadly the offset of one hour is still present. my Configuration.yaml contains the correct latitude and longitude and and the timezone is properly set there too to "Europe/Amsterdam". :(

**Comment by craigbarratt** [comment]
Sorry, I can't help if it's not possible to run some simple tests using vanilla python on your platform.

**Comment by dlashua** [comment]
Setting your timezone in Home Assistant does not set it for the entire system. My system timezone is set correctly so I don't have to worry about this.

There's probably a way to pull your timezone information from the Home Assistant config and use it when generating dates, but I'm not sure what it is. It would be easier to just hardcode it.

```
from datetime import datetime
import pytz

my_tz = pytz.timezone('Europe/Amsterdam')

my_datetime = datetime.now(my_tz)

log.info(my_datetime.strftime("%H:%M:%S")
```

**Reply by Danahi** [reply]
Somehow it rectified itself... The line that you suggested prints the correct time. And the @time_trigger("once(<in one minute>)") also triggers.
Sadly no clear reason why. I did fix something in the configuration.yaml, perhaps that invalidated the timezone?
